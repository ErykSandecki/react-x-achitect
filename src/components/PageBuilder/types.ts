// types
import { TRectCoordinates } from 'types';

export type TRectArea = TRectCoordinates | null;
