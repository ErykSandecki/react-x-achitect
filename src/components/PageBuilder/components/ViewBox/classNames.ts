export const className = 'ViewBox';

export const classNames = {
  [className]: {
    name: className,
    modificators: {
      createFrame: `${className}--create-frame`,
    },
  },
};
