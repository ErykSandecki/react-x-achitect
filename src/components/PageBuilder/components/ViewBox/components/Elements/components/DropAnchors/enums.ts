export enum DropAnchorsPosition {
  top = 'top',
  bottom = 'bottom',
  left = 'left',
  right = 'right',
}
