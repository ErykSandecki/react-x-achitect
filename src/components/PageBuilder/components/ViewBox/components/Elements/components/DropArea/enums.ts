export enum DropAreaPosition {
  top = 'top',
  bottom = 'bottom',
  left = 'left',
  right = 'right',
}
