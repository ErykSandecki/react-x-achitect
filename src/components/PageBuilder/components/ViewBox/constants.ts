// others
import { translationNameSpace as parentNameSpace } from '../../constants';

export const OVERLAY_CONTAINER_ID = 'overlayContainer';
export const THROTTLE_WAIT = 20;
export const translationNameSpace = `${parentNameSpace}.viewBox`;
