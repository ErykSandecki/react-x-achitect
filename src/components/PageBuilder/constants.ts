export const translationNameSpace = 'pageBuilder';

export const TOOLBAR_HEIGHT = 56;
