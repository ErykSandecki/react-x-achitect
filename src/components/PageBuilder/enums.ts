export enum MouseMode {
  menu = 'menu',
  default = 'default',
  move = 'move',
  comment = 'comment',
  toolBeltA = 'toolBeltA',
}
