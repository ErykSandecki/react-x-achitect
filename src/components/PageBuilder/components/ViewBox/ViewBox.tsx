import { FC, useRef } from 'react';

// components
import Elements from './components/Elements/Elements';
import FrameArea from './components/FrameArea/FrameArea';
import { ZoomBox } from 'shared';

// others
import { className, classNames } from './classNames';

// hooks
import { useTheme } from 'hooks';
import { useViewBoxEvents } from './hooks/useViewBoxEvents';

// styles
import styles from './view-box.scss';

// types
import { MouseMode } from 'components/PageBuilder/enums';
import { T3DCoordinates } from 'types';

export type TViewBoxProps = {
  coordinates: T3DCoordinates;
  mouseMode: MouseMode;
  setCoordinates: (coordinates: T3DCoordinates) => void;
  setMouseMode: (mouseMode: MouseMode) => void;
};

const ViewBox: FC<TViewBoxProps> = ({
  coordinates,
  mouseMode,
  setCoordinates,
  setMouseMode,
}) => {
  const zoomBoxRef = useRef<HTMLDivElement>(null);
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);
  const { frameArea, ...events } = useViewBoxEvents(
    coordinates,
    mouseMode,
    setMouseMode,
  );
  const eventsDisabled = frameArea !== null || mouseMode !== MouseMode.default;

  return (
    <ZoomBox
      classes={{
        className: cx(classNamesWithTheme[className].name, [
          classNamesWithTheme[className].modificators.createFrame,
          mouseMode === MouseMode.toolBeltA,
        ]),
      }}
      coordinates={coordinates}
      onMouseMoveDepedencies={[frameArea, mouseMode]}
      onMouseUpDepedencies={[frameArea, mouseMode]}
      setCoordinates={setCoordinates}
      zoomBoxRef={zoomBoxRef}
      {...events}
    >
      <Elements eventsDisabled={eventsDisabled} />
      <FrameArea frameArea={frameArea} />
    </ZoomBox>
  );
};

export default ViewBox;
