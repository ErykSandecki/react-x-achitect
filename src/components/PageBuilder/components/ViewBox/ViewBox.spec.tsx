import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

// components
import ViewBox from './ViewBox';

// mocks
import 'test/mocks/sagas/allSagas';

// others
import { BASE_3D } from 'shared';

// store
import { configureStore } from 'store/store';

// types
import { MouseMode } from 'components/PageBuilder/enums';

const mockCallBack = jest.fn();

describe('ViewBox snapshots', () => {
  it('should render ViewBox', () => {
    // mock
    const store = configureStore();

    // before
    const { asFragment } = render(
      <Provider store={store}>
        <ViewBox
          coordinates={BASE_3D}
          mouseMode={MouseMode.default}
          setCoordinates={mockCallBack}
          setMouseMode={mockCallBack}
        />
      </Provider>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
