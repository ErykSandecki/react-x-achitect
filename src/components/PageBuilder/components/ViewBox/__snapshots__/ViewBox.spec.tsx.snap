// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewBox snapshots should render ViewBox 1`] = `
<DocumentFragment>
  <div
    class="css-17c89wm ViewBox ViewBox--light ZoomBox ZoomBox--light ZoomBox__idle ZoomBox__idle--light ZoomBox__default ZoomBox__default--light"
    data-test-box="zoom-box"
  >
    <div
      class="css-qel697 ZoomBox__zoom-content ZoomBox__zoom-content--light"
      data-test-box=""
      style="transform: translate(0px, 0px) scale(1);"
    />
  </div>
</DocumentFragment>
`;
